src/stdlib/IntUtils.scilla

```diff
@@ -0,0 +1,104 @@
+library IntUtils
+
+let int_eq =
+  tfun 'A =>
+  fun (a : 'A) =>
+  fun (b : 'A) =>
+    builtin eq a b
+
+let int_neq =
+  tfun 'A =>
+  fun (a : 'A) =>
+  fun (b : 'A) =>
+    let eq = builtin eq a b in
+    match eq with
+    | True => False
+    | False => True
+    end
+
+let int_lt =
+  tfun 'A =>
+  fun (a : 'A) =>
+  fun (b : 'A) =>
+    builtin lt a b
+
+let int_le =
+  tfun 'A =>
+  fun (a : 'A) =>
+  fun (b : 'A) =>
+    let lt = builtin lt a b in
+    match lt with
+    | True => True
+    | False => builtin eq a b
+    end
+
+let int_gt =
+  tfun 'A =>
+  fun (a : 'A) =>
+  fun (b : 'A) =>
+    builtin lt b a
+
+let int_ge =
+  tfun 'A =>
+  fun (a : 'A) =>
+  fun (b : 'A) =>
+    let gt = builtin lt b a in
+    match gt with
+    | True => True
+    | False => builtin eq b a
+    end
```

Comment:
Could this be more elegantly defined as `@int_le 'A b a`?

Reply:
Done.